backend/app/tests/integration/user_test.py

```diff
@@ -0,0 +1,187 @@
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.data)
+        self.assertIn('Users retrieved successfully', data['message'])
+
+    def test_get_users_as_regular_user(self):
+        """Test getting user list as a regular user (should fail)"""
+        self.test_user_login()
+        headers = {'Authorization': f'Bearer {self.user_access_token}'}
+        response = self.client.get('/users', headers=headers)
+        self.assertEqual(response.status_code, 403)
+
+    def test_create_user_as_admin(self):
+        """Test creating a new user as an admin"""
+        headers = self.get_auth_headers(self.admin_user_id)
+        response = self.client.post('/users', headers=headers, json={
+            'username': 'newuser',
+            'password': 'newpass',
+            'first_name': 'New',
+            'last_name': 'User',
+            'email': 'newuser@example.com',
+            'role': 'regular'
+        })
+        self.assertEqual(response.status_code, 201)
+        data = json.loads(response.data)
+        self.assertIn('User created successfully', data['message'])
+
+    def test_create_user_as_regular_user(self):
+        """Test creating a new user as a regular user (should fail)"""
+        self.test_user_login()
+        headers = {'Authorization': f'Bearer {self.user_access_token}'}
+        response = self.client.post('/users', headers=headers, json={
+            'username': 'anotheruser',
+            'password': 'anotherpass',
+            'first_name': 'Another',
+            'last_name': 'User',
+            'email': 'anotheruser@example.com',
+            'role': 'regular'
+        })
+        self.assertEqual(response.status_code, 403)
+
+    def test_get_user_by_id_as_admin(self):
+        """Test getting a user by ID as an admin"""
+        headers = self.get_auth_headers(self.admin_user_id)
+        response = self.client.get(f'/users/{self.admin_user_id}', headers=headers)
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.data)
+        self.assertIn('User retrieved successfully', data['message'])
+
+    def test_delete_user_as_admin(self):
+        """Test deleting a user as an admin"""
+        # Create a new user to delete
+        headers = self.get_auth_headers(self.admin_user_id)
+        response = self.client.post('/users', headers=headers, json={
+            'username': 'todelete',
+            'password': 'deletepass',
+            'first_name': 'To',
+            'last_name': 'Delete',
+            'email': 'todelete@example.com',
+            'role': 'regular'
+        })
+        data = json.loads(response.data)
+        user_id = data['payload']['id']
+        # Delete the user
+        response = self.client.delete(f'/users/{user_id}', headers=headers)
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.data)
+        self.assertIn('User deleted successfully', data['message'])
```

Comment:
Separate your tests into different files(define your test case an abstract class and let other files extend it).

- [ ] refactor as it's very difficult to review in its current form